Assets/GothicVR/Scripts/Interactive/Seat.cs

```diff
@@ -0,0 +1,129 @@
+            GameObject player = args.interactorObject.transform.GetComponentInParent<XROrigin>().gameObject;
+            //handle sitting/standing
+            isPlayerSeated = !isPlayerSeated;
+            if (isPlayerSeated) StartCoroutine(SitDown(player));
+            else StartCoroutine(StandUp(player));
+            Invoke("EnableSitting", sittingCooldown);
+        }
+
+        private void EnableSitting()
+        {
+            canPlayerSit = true;
+        }
+
+        private IEnumerator SitDown(GameObject player)
+        {
+            //cache player pos and eulers
+            cachedPos = player.transform.position;
+            cachedEulers = player.transform.eulerAngles;
+
+            //lock input
+            cachedLocomotion = player.GetComponentInChildren<LocomotionSystem>().gameObject;
+            cachedLocomotion.SetActive(false);
+
+            //get snap point
+            currentSnapPoint = GetNearestSnapPoint(player.transform.position);
+
+            //fade camera out
+            cachedCameraFade = player.GetComponentInChildren<CameraFade>();
+            cachedCameraFade.Fade(cameraFadeDuration, 1);
+            yield return new WaitForSeconds(cameraFadeDuration);
+
+            //set position and rotation
+            player.transform.position = currentSnapPoint.position + currentSnapPoint.TransformDirection(posOffset);
+            player.transform.eulerAngles = currentSnapPoint.eulerAngles + eulerOffset;
+
+            //fade camera in
+            cachedCameraFade.Fade(cameraFadeDuration, 0);
+        }
+        private IEnumerator StandUp(GameObject player)
+        {
+            //fade camera out
+            cachedCameraFade.Fade(cameraFadeDuration, 1);
+            yield return new WaitForSeconds(cameraFadeDuration);
+            //move player forward
+            player.transform.position += player.transform.TransformDirection(new Vector3(0, 0.5f, 1f));
+           // player.transform.eulerAngles = cachedEulers;
+            //unlock move an turn input
+            cachedLocomotion.SetActive(true);
+            //fade camera in
+            cachedCameraFade.Fade(cameraFadeDuration, 0);
+            //clear cache
+            cachedPos = Vector3.zero;
+            cachedEulers = Vector3.zero;
+            cachedLocomotion = null;
+            cachedCameraFade = null;
+        }
+        private Transform GetNearestSnapPoint(Vector3 pos)
+        {
+            //find the closest Snap Point for player to sit
+            Transform nearestSnapPoint = null;
+            float minDistance = float.MaxValue;
+            for (int i = 0; i < snapPoints.Count; i++)
+            {
+                float distance = Vector3.Distance(pos, snapPoints[i].position);
+                if (distance < minDistance)
+                {
+                    minDistance = distance;
+                    nearestSnapPoint = snapPoints[i];
+                }
+            }
+            return nearestSnapPoint;
+        }
```

Comment:
You could even use LINQ:
```suggestion
        private Transform GetNearestSnapPoint(Vector3 pos)
        {
            // Find the closest Snap Point for player to sit.
            return snapPoints
                .OrderBy(point => Vector3.Distance(pos, point.position))
                .First();
        }
```
Beautiful isn't it. :-D

Reply:
fair shout!